crates/biome_cli/src/logging.rs

```diff
@@ -44,14 +44,15 @@ pub fn setup_cli_subscriber(
         .with_span_events(fmt_span)
         .with_writer(make_writer);
 
-    let layer = match kind {
-        LoggingKind::Pretty => layer.pretty().first(),
-        LoggingKind::Compact => layer.compact().second(),
-        LoggingKind::Json => layer.json().flatten_event(true).third(),
-    }
-    .with_filter(LoggingFilter { level });
-
-    registry().with(layer).init();
+    let registry = registry();
+    let filter = LoggingFilter { level };
+    match kind {
+        LoggingKind::Pretty => registry.with(layer.pretty().with_filter(filter)).init(),
+        LoggingKind::Compact => registry.with(layer.compact().with_filter(filter)).init(),
+        LoggingKind::Json => registry
+            .with(layer.json().flatten_event(true).with_filter(filter))
+            .init(),
+    };
```

Comment:
What are we fixing here? a regression? would be nice to add it to the PR description and add a changeset for it

Reply:
I couldn't fix this without the logs working so I fixed the logs. I documented this in the commit message, it is perhaps better to review my code commit by commit.